### PR TITLE
Minor cleanup after #469

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
This is mostly a nonfunctional change PR.  See changes below

## Changes proposed in this PR:
- Re-enable a GDP test disabled by @whart222 in #469 
- Minor PEP8 and documentation cleanup.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/core/expr/numvalue.py [hunk 1]
```diff
@@ -384,6 +413,7 @@ def as_numeric(obj):
         val = _KnownConstants.get(obj, None)
         if val is not None:
             return val 
+        print ".",
```
Comment (michaelbynum) [inline pyomo/core/expr/numvalue.py]: @jsiirola I assume this print statement was supposed to be deleted?

File: pyomo/core/expr/numvalue.py [hunk 2]
```diff
@@ -272,9 +286,14 @@ def is_fixed(obj):
     try:
         # Now we need to confirm that we have an unknown numeric type
         check_if_numeric_type_and_cache(obj)
+        # As this branch is only hit for previously unknown (to Pyomo)
+        # types that behave reasonably like numbers, we know that *must*
```
Comment (blnicho) [inline pyomo/core/expr/numvalue.py]: Typo: "we know that **they** must be fixed"

File: pyomo/core/expr/numvalue.py [hunk 3]
```diff
@@ -320,6 +339,9 @@ def is_numeric_data(obj):
...
```
Comment (blnicho) [inline pyomo/core/expr/numvalue.py]: Typo: "we know that **they** must be"

File: pyomo/core/expr/numvalue.py [hunk 4]
```diff
@@ -334,31 +356,38 @@ def polynomial_degree(obj):
...
+            % (type(obj).__name__,))
     try:
         return obj.polynomial_degree()
     except AttributeError:
         pass
     try:
         # Now we need to confirm that we have an unknown numeric type
         check_if_numeric_type_and_cache(obj)
+        # As this branch is only hit for previously unknown (to Pyomo)
+        # types that behave reasonably like numbers, we know that *must*
```
Comment (blnicho) [inline pyomo/core/expr/numvalue.py]: Typo: "we know that **they** must be"

File: pyomo/core/expr/numvalue.py (hunk 1 above)
Comment (jsiirola) [reply pyomo/core/expr/numvalue.py]: Yup.

File: pyomo/core/expr/numvalue.py [hunk 5]
```diff
@@ -334,31 +356,38 @@ def polynomial_degree(obj):
...
-#   For now, all constants are coerced to floats.  This avoids integer division
-#   in Python 2.x.  (At least some of the time.)
+#   For now, all constants are coerced to floats.  This avoids integer
+#   division in Python 2.x.  (At least some of the time.)
 #
-#   When we eliminate support for Python 3.x, we will not need this coercion.  The main
-#   difference in the following code is that we will need to index
-#   KnownConstants by both the class type and value, since INT, FLOAT and LONG values
-#   sometimes hash the same.
+#   When we eliminate support for Python 3.x, we will not need this
```
Comment (blnicho) [inline pyomo/core/expr/numvalue.py]: Typo: "eliminate support for Python **2.x**"

File: pyomo/core/expr/numvalue.py (hunk 1 above)
Comment (blnicho) [inline pyomo/core/expr/numvalue.py]: Rogue print statement?

File: pyomo/core/expr/numvalue.py [hunk 6]
```diff
@@ -398,9 +428,9 @@ def as_numeric(obj):
         #
         retval = NumericConstant(obj)
         #
-        # Cache the numeric constants.  We used a bounded the cache size to avoid
-        # unexpectedly large lists of constants.  There are typically a small number
-        # of constants that need to be cached.
+        # Cache the numeric constants.  We used a bounded the cache size
```
Comment (blnicho) [inline pyomo/core/expr/numvalue.py]: Typo: "used a bounded cache size"

**Pull request merged**